sdk/storage/Azure.Storage.Common/tests/AuthenticatedRegionCryptoStreamTest.cs

```diff
@@ -0,0 +1,432 @@
+                    Enumerable.Repeat(_tagByte, _tagLength),
+                    ciphertextAuthBlock.Slice(ciphertextAuthBlock.Length - _tagLength).ToArray());
+            }
+        }
+
+        [Test]
+        [Combinatorial]
+        public void TransformDecryptRead(
+            [Values(true, false)] bool alligned,
+            [Values(1, 3)] int numAuthBlocks,
+            [Values(200, _authRegionDataLength, null)] int? streamReadLength)
+        {
+            // Arrange
+            int ciphertextLength = (alligned ? _totalAuthRegionLength : 500) + ((numAuthBlocks - 1) * _totalAuthRegionLength);
+            ReadOnlySpan<byte> ciphertext = new ReadOnlySpan<byte>(GetRandomBytes(ciphertextLength));
+
+            var readStream = new AuthenticatedRegionCryptoStream(
+                new MemoryStream(ciphertext.ToArray()),
+                new MockDecryptTransform(_nonceLength, _tagLength),
+                _authRegionDataLength,
+                CryptoStreamMode.Read);
+
+            // Act
+            int read;
+            int totalRead = 0;
+            byte[] plaintextResult = new byte[ciphertextLength - (numAuthBlocks * (_nonceLength + _tagLength))];
+            do
+            {
+                read = streamReadLength.HasValue
+                    ? readStream.Read(plaintextResult, totalRead, Math.Min(streamReadLength.Value, plaintextResult.Length - totalRead))
+                    : readStream.Read(plaintextResult, totalRead, plaintextResult.Length - totalRead);
+                totalRead += read;
+            } while (read != 0);
+
+            // Assert
+            Assert.AreEqual(plaintextResult.Length, totalRead);
+            foreach (int authBlock in Enumerable.Range(0, numAuthBlocks))
+            {
+                int plaintextOffset = authBlock * _authRegionDataLength;
+                ReadOnlySpan<byte> plaintextAuthBlock = new ReadOnlySpan<byte>(
+                    plaintextResult,
+                    plaintextOffset,
+                    Math.Min(_authRegionDataLength, plaintextResult.Length - plaintextOffset));
+
+                int ciphertextOffset = authBlock * _totalAuthRegionLength;
+                ReadOnlySpan<byte> ciphertextAuthBlock = ciphertext.Slice(
+                    ciphertextOffset,
+                    Math.Min(_totalAuthRegionLength, ciphertext.Length - ciphertextOffset));
+
+                CollectionAssert.AreEqual(
+                    plaintextAuthBlock.ToArray(),
+                    ciphertextAuthBlock.Slice(
+                        _nonceLength,
+                        Math.Min(_authRegionDataLength, ciphertextAuthBlock.Length - _nonceLength - _tagLength)).ToArray());
+            }
+        }
+
+#if NETCOREAPP3_1
+        [Test]
+        [Combinatorial]
+        public void RoundTrip_Read(
+            [Values(true, false)] bool useRealEncryption,
+            [Values(true, false)] bool alligned,
+            [Values(1, 3)] int numAuthBlocks)
```

Comment:
@amnguye this is the one test that needs the unsafe tag to be added, and only for `useRealEncryption == true`. I'm not sure just removing a test is a good idea, but I'll admit it becomes more of an integration test than a unit test when we use GCM here. Perhaps we need a broader effort to classify tests as unit/integration/smoke. This would likely cut down on bulky test classes and packages in general.

Reply:
I have moved it to the blob encryption tests, which already need to be configured for something like this. I can take a work item later to revisit our testing organization.